### ISSUE TITLE
Integrate WooPayments test drive mode into the onboarding flow

### ISSUE THREAD

**Issue by mordeth** [Automattic/woocommerce-payments#9140]
### Description
Integrate a test-driving mode in WooPayments onboarding flow that closely replicates the production environment. The solution should provide a one-click onboarding for custom test accounts eliminating all irrelevant factors except Jetpack connection.

### Acceptance criteria
The implementation should replace the existing Sandbox mode and provide one-click onboarding for test-drive accounts, with the exception of the Jetpack connection which cannot currently be bypassed. This should demonstrate all the features of a fully onboarded production account, except for processing live payments.

After successful onboarding, the merchant should be redirected to the Payments Overview page with a fully onboarded test account.

### Design
In progress

### Additional context
One-pager: paJDYF-cMY-p2
Spike POC: paJDYF-dZe-p2

**Comment by timmy5685** [comment]
@mordeth here are the final designs for this flow:

New Loading screen
![image](https://github.com/user-attachments/assets/2a75f16e-a5a3-4cc7-9536-8a83760fc60d)
Figma: VLWDmeuIXQ7XAK4flyqTFO-fi-3101_6517

New snack bar and _updated_ text in notification at the top of the screen
![image](https://github.com/user-attachments/assets/5b0901ed-19a5-4f70-b8c0-dafa8797834c)
Figma: VLWDmeuIXQ7XAK4flyqTFO-fi-3101_6517

### Expected flow:
**No jetpack connection exists**
1. Merchant clicks enable sandbox mode
2. Merchant is directed to jetpack
3. Merchant is directed to new loading screen
4. Merchant is directed to payments overview page with updated notification text and snack bar message

**Jetpack connection already exists**
1. Merchant clicks enable sandbox mode
3. Merchant is directed to new loading screen
4. Merchant is directed to payments overview page with updated notification text and snack bar message

cc: @orcunomattic

**Comment by timmy5685** [comment]
@mordeth - apologies but I was just looking at the Pull request and noticed that the copy for notification in the PR was incorrect. I went back here and checked and realized that was my fault. I called out the updated copy but uploaded the wrong screenshot.

This is the correct one - it should say "real payments" instead of real transactions.
<img width="1440" alt="Payments Overview" src="https://github.com/user-attachments/assets/408379fa-6d38-4700-a361-8ab938f80932">
